File: zero judge/a006 一元二次方程式.cpp
#include <iostream>
#include <cmath>
using namespace std;
int main(){	
    int a,b,c,x,y;
    cin>>a>>b>>c;
    if(b*b-4*a*c==0){
    x=(-b+(sqrt(b*b-4*a*c)))/(2*a);
    cout<<"Two same roots "<<"x="<<x<<endl;}
	else if(b*b-4*a*c>0)
    {x=(-b+(sqrt(b*b-4*a*c)))/2*a;
    y=(-b-(sqrt(b*b-4*a*c)))/2*a;
    cout<<"Two different roots "<<"x1="<<x<<" , "<<"x2="<<y<<endl;}
	else if((b*b-4*a*c)<0)
	{
    cout<<"No real root"<<endl;
}
	return 0;
}

File: zero judge/a006. 一元二次方程式.py
import math
a,b,c = map(int,input().split())
sol = b ** 2 - 4 * a * c
if sol == 0:
	print("Two same roots x=",-b // (2 * a),sep='')
elif sol > 0:
	print("Two different roots x1=",(-b + int(math.sqrt(b ** 2 - 4 * a * c))) // (2 * a)," , x2=",(-b - int(math.sqrt(b ** 2 - 4 * a * c))) // (2 * a),sep='')
else:
	print("No real root")